metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -0,0 +1,1152 @@
+        capabilities.setCallHierarchyProvider(true)
+        capabilities.setWorkspaceSymbolProvider(true)
+        capabilities.setDocumentSymbolProvider(true)
+        capabilities.setDocumentFormattingProvider(true)
+        if (initializeParams.supportsCodeActionLiterals) {
+          capabilities.setCodeActionProvider(
+            new lsp4j.CodeActionOptions(
+              List(
+                lsp4j.CodeActionKind.QuickFix,
+                lsp4j.CodeActionKind.Refactor,
+                lsp4j.CodeActionKind.SourceOrganizeImports,
+              ).asJava
+            )
+          )
+        } else {
+          capabilities.setCodeActionProvider(true)
+        }
+
+        val textDocumentSyncOptions = new lsp4j.TextDocumentSyncOptions
+        textDocumentSyncOptions.setChange(lsp4j.TextDocumentSyncKind.Full)
+        textDocumentSyncOptions.setSave(new lsp4j.SaveOptions(true))
+        textDocumentSyncOptions.setOpenClose(true)
+
+        val scalaFilesPattern = new lsp4j.FileOperationPattern("**/*.scala")
+        scalaFilesPattern.setMatches(lsp4j.FileOperationPatternKind.File)
+        val folderFilesPattern = new lsp4j.FileOperationPattern("**/")
+        folderFilesPattern.setMatches(lsp4j.FileOperationPatternKind.Folder)
+        val fileOperationOptions = new lsp4j.FileOperationOptions(
+          List(
+            new lsp4j.FileOperationFilter(scalaFilesPattern),
+            new lsp4j.FileOperationFilter(folderFilesPattern),
+          ).asJava
+        )
+        val fileOperationsServerCapabilities =
+          new lsp4j.FileOperationsServerCapabilities()
+        fileOperationsServerCapabilities.setWillRename(fileOperationOptions)
+        val workspaceCapabilitiesOptions = new lsp4j.WorkspaceFoldersOptions()
+        workspaceCapabilitiesOptions.setSupported(true)
+        workspaceCapabilitiesOptions.setChangeNotifications(true)
+        val workspaceCapabilities =
+          new lsp4j.WorkspaceServerCapabilities(workspaceCapabilitiesOptions)
+        workspaceCapabilities.setFileOperations(
+          fileOperationsServerCapabilities
+        )
+        capabilities.setWorkspace(workspaceCapabilities)
+
+        capabilities.setTextDocumentSync(textDocumentSyncOptions)
+
+        val gson = new Gson
+        val data =
+          gson.toJsonTree(MetalsExperimental())
+        capabilities.setExperimental(data)
+        val serverInfo = new lsp4j.ServerInfo("Metals", BuildInfo.metalsVersion)
+        new lsp4j.InitializeResult(capabilities, serverInfo)
+      })
+      .asJava
+  }
+
+  def cancel(): Unit = {
+    cancelables.cancel()
+    folderServices.foreach(_.cancel())
+  }
+
+  def initialized(): Future[Unit] = {
+    statusBar.start(sh, 0, 1, ju.concurrent.TimeUnit.SECONDS)
+    folderServices.foreach { service =>
+      service.registerNiceToHaveFilePatterns()
+      service.connectTables()
+    }
+    syncUserconfiguration().flatMap { _ =>
+      Future // TODO:: we should probably have only one http server
```

Comment:
I think that would be best if possible

Reply:
I'd rather do that as a follow up, would that be okay? That would also allow/require having a single doctor page that collects all the workspace data.